src/main/java/de/ait/secondlife/services/OfferServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+import de.ait.secondlife.domain.dto.OfferResponseDto;
+import de.ait.secondlife.domain.dto.OfferResponseWithPaginationDto;
+import de.ait.secondlife.domain.dto.OfferUpdateDto;
+import de.ait.secondlife.domain.entity.Offer;
+import de.ait.secondlife.exception_handling.exceptions.badRequestException.CreateOfferConstraintViolationException;
+import de.ait.secondlife.exception_handling.exceptions.badRequestException.isNullExceptions.IdIsNullException;
+import de.ait.secondlife.exception_handling.exceptions.notFoundException.OfferNotFoundException;
+import de.ait.secondlife.exception_handling.exceptions.badRequestException.WrongAuctionParameterException;
+import de.ait.secondlife.repositories.OfferRepository;
+import de.ait.secondlife.services.interfaces.OfferService;
+import de.ait.secondlife.services.interfaces.StatusService;
+import de.ait.secondlife.services.mapping.OfferMappingService;
+import jakarta.transaction.Transactional;
+import jakarta.validation.ConstraintViolationException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+
+@Service
+@RequiredArgsConstructor
+public class OfferServiceImpl implements OfferService {
+
+    private final OfferRepository offerRepository;
+    private final OfferMappingService mappingService;
+    private final StatusService statusSevice;
+
+    @Override
+    public OfferResponseWithPaginationDto findOffers(Pageable pageable) {
+        Page<Offer> pageOfOffer = offerRepository.findAllByIsActiveTrue(pageable);
+        return offersToOfferRequestWithPaginationDto(pageOfOffer);
+    }
+
+    @Override
+    public OfferResponseDto findOfferById(UUID id) {
+        if (id == null) throw new IdIsNullException();
+        Offer offer = offerRepository.findByIdAndIsActiveTrue(id)
+                .orElseThrow(() -> new OfferNotFoundException(id));
+        return mappingService.toRequestDto(offer);
+    }
+
+    @Override
+    public OfferResponseWithPaginationDto findOffersByUserId(Long id, Pageable pageable) {
+        if (id == null) throw new IdIsNullException();
+        Page<Offer> pageOfOffer = offerRepository.findByUserIdAndIsActiveTrue(id, pageable);
+        return offersToOfferRequestWithPaginationDto(pageOfOffer);
+    }
+
+    @Override
+    public OfferResponseDto createOffer(OfferCreationDto dto) {
+        try {
+            Offer newOffer = mappingService.toOffer(dto);
+            newOffer.setId(null);
+            newOffer.setStatus(statusSevice.getStatusByName(OfferStatus.DRAFT.name()));
+            newOffer.setAuctionDurationDays(newOffer.getAuctionDurationDays() <= 0 ? 3 : newOffer.getAuctionDurationDays());
+            if (dto.getIsFree() == null) {
```

Comment:
unnecessary check, since there is a limitation in the DTO
_@NotNull(message = "Is free cannot be null")
private Boolean isFree;_